samples/Gonda/Gonda.Server/Controllers/AuthorizationController.cs

```diff
@@ -0,0 +1,132 @@
+using static OpenIddict.Abstractions.OpenIddictConstants;
+
+namespace Gonda.Server.Controllers
+{
+    public class AuthorizationController : Controller
+    {
+        private readonly SignInManager<IdentityUser> _signInManager;
+        private readonly UserManager<IdentityUser> _userManager;
+        private readonly IOpenIddictScopeManager _scopeManager;
+
+        public AuthorizationController(
+            SignInManager<IdentityUser> signInManager,
+            UserManager<IdentityUser> userManager,
+            IOpenIddictScopeManager scopeManager)
+        {
+            _signInManager = signInManager;
+            _userManager = userManager;
+            _scopeManager = scopeManager;
+        }
+
+        [HttpPost("~/connect/token"), Produces("application/json")]
+        public async Task<IActionResult> Exchange()
+        {
+            var request = HttpContext.GetOpenIddictServerRequest();
+            if (request.IsPasswordGrantType())
+            {
+                var user = await _userManager.FindByNameAsync(request.Username);
+                if (user == null)
+                {
+                    var properties = new AuthenticationProperties(new Dictionary<string, string>
+                    {
+                        [OpenIddictServerAspNetCoreConstants.Properties.Error] = Errors.InvalidGrant,
+                        [OpenIddictServerAspNetCoreConstants.Properties.ErrorDescription] =
+                            "The username/password couple is invalid."
+                    });
+
+                    return Forbid(properties, OpenIddictServerAspNetCoreDefaults.AuthenticationScheme);
+                }
+
+                // Validate the username/password parameters and ensure the account is not locked out.
+                var result =
+                    await _signInManager.CheckPasswordSignInAsync(user, request.Password, lockoutOnFailure: true);
+                if (!result.Succeeded)
+                {
+                    var properties = new AuthenticationProperties(new Dictionary<string, string>
+                    {
+                        [OpenIddictServerAspNetCoreConstants.Properties.Error] = Errors.InvalidGrant,
+                        [OpenIddictServerAspNetCoreConstants.Properties.ErrorDescription] =
+                            "The username/password couple is invalid."
+                    });
+
+                    return Forbid(properties, OpenIddictServerAspNetCoreDefaults.AuthenticationScheme);
+                }
+
+                // Create a new ClaimsPrincipal containing the claims that
+                // will be used to create an id_token, a token or a code.
+                var principal = await _signInManager.CreateUserPrincipalAsync(user);
+
+                var scopes = request.GetScopes();
+                
+                principal.SetScopes(request.GetScopes());
+                // principal.SetResources(await _scopeManager.ListResourcesAsync(scopes).ToListAsync());
```

Comment:
Any particular reason for commenting this call?

Reply:
I think I'd removed this while testing something. I just uncommented it and the demo is still working